src/java.base/unix/native/libjava/ProcessImpl_md.c

```diff
@@ -770,7 +806,7 @@ Java_java_lang_ProcessImpl_forkAndExec(JNIEnv *env,
         throwIOException(env, errnum, "Exec failed");
```

Comment:
Why do we need to keep `throwIOException()` for this single call site? Can't we replace it with `throwInternalIOException()` as well?

Otherwise looks good.

Reply:
Because this is the path through which we report "normal" errors like "command not found", "path does not exist", etc. -- that are not related to jspawnhelper failures at all. So it would be dubious to print the warnings then. I added a few test cases in `Basic.java` that checks for `IOException` messages int those cases.